### PR TITLE
feat(dataverse-integration): add base classes and templates

### PR DESCRIPTION
**Pull request by jmurugan-fzj**
- Added the base UI files along with the extensions for the dataverse integration feature. Just added the basic UI files and the base classes with minimal/dummy functionality
- Added the backend thread-task extension for concurrent uploads in the background with an upload manager
- Added base database API for creating/retrieving documents/design documents/views etc. along with an extension for dataverse operations and database models

### CODE REVIEW

**Comment by jmurugan-fzj** [comment]
@SteffenBrinckmann Could you please review this one, just all the basic classes needed are added even though the functionality is not as fully required, once the base classes are in, I will implement each functionality as separate PRs..

File: .github/workflows/installLinux.yml
[the remark this answers is not in the document]
Comment (jmurugan-fzj) [reply .github/workflows/installLinux.yml:1]: @SteffenBrinckmann Without which this leads to the same error (github action "Verify Linux install / build ") what you have notice as part of your PR #195

**Comment by jmurugan-fzj** [comment]
@SteffenBrinckmann Could you review and help me finalize this PR soon?

File: .github/workflows/installLinux.yml
[the remark this answers is not in the document]
Comment (jmurugan-fzj) [reply .github/workflows/installLinux.yml:1]: @SteffenBrinckmann Sorry, I had removed all the un-necessary libs and included only Mesa 3D Graphics library which is needed by both the actions..

**Comment by SteffenBrinckmann** [comment]
@jmurugan-fzj could you look at the other two comments, that I had asked. A short answer suffices.

**Comment by jmurugan-fzj** [comment]
> @jmurugan-fzj could you look at the other two comments, that I had asked. A short answer suffices.

@SteffenBrinckmann I see only one comment regarding the libs, are you sure you posted it or still in review?

File: pasta_eln/dataverse/data_upload_task.py
[the remark this answers is not in the document]
Comment (jmurugan-fzj) [reply pasta_eln/dataverse/data_upload_task.py:1]: This only signals the UI with the data updates, does not directly do anything with UI or manage the UI items, only for signals, it's requires QT

**Pull request merged**